### PR TITLE
Add support for uploading multiple release artifacts to the same edit

### PR DESCRIPTION
**Pull request by boswelja**
Useful for projects that target multiple platforms (Wear OS, Android TV, Android phones etc).
Needs more testing, but I've been using these changes for one of my projects and uploading 2 artifacts at a time seems to work fine.

### CODE REVIEW

**Comment by boswelja** [comment]
This will resolve #29

**Comment by r0adkll** [comment]
This is a pretty large change so it's going to take me some time to verify and work through it.

**Comment by boswelja** [comment]
While I've had a successful publish via these changes before, I seem to have started getting errors now. Investigating.

**Comment by boswelja** [comment]
Commented code a bit to better show what uploadToPlayStore does now.
The only regression I'm aware of is that internal sharing artifacts don't get their download URL  stored in actions output, not sure how to go about this. Thoughts?

File: src/edits.ts [hunk 1]
```diff
@@ -28,89 +28,113 @@ export interface EditOptions {
...
+        
+        // Validate the given track
+        await validateSelectedTrack(appEdit.data, options).catch(reason => {
+            core.setFailed(reason);
+            return Promise.reject(reason);
+        });
+    
+        // Upload artifacts to Google Play, and store their version codes
+        var versionCodes = new Array<number>();
+        releaseFiles.forEach(async releaseFile => {
```
Comment (r0adkll) [inline src/edits.ts:56]: You are going to run into issues with this block (and similar others) where you are looping and spinning off async tasks without blocking for them so the next blocks of code (add releases, committing edit) are going to potentially execute without this finishing. You should refer to this post (https://stackoverflow.com/a/37576787) for writing these loops in a blocking fashion.

File: src/edits.ts [hunk 2]
```diff
@@ -28,89 +28,113 @@ export interface EditOptions {
...
-    } else {
-        core.setFailed(`Error ${res.status}: ${res.statusText}`)
-        return Promise.reject(res.status)
-    }
-}
-
-async function uploadRelease(appEdit: AppEdit, options: EditOptions, releaseFile: string): Promise<string | undefined | null> {
     // Check the 'track' for 'internalsharing', if so switch to a non-track api
-    if (options.track === 'internalsharing'){
+    if (options.track === 'internalappsharing') {
```
Comment (r0adkll) [inline src/edits.ts:32]: Please leave this as `internalsharing` or update the README.md file to reflect this change. I would prefer to leave it as is to keep compatibility with users already using this track.

File: README.md [hunk 3]
```diff
@@ -18,9 +18,9 @@ The service account json in plain text, provided via a secret, etc.
 
 **Required:** The package name, or Application Id, of the app you are uploading
 
-### `releaseFile`
+### `releaseFiles`
```
Comment (r0adkll) [inline README.md:21]: Please update the `actions.yml` file to reflect this input change

**Review by r0adkll — CHANGES_REQUESTED** [review]

**Comment by r0adkll** [comment]
> The only regression I'm aware of is that internal sharing artifacts don't get their download URL stored in action output, not sure how to go about this. Thoughts?

Yeah, this is another outstanding issue that I'm trying to wrap my head around. My current theory is something to do with the service account used to authenticate these requests not matching approved app sharing uploads and therefore doesn't return a URL to share. I would leave this out of this PR for now in favor of working on it separately since it is out of scope for this change.

File: src/edits.ts (hunk 1 above)
Comment (boswelja) [reply src/edits.ts:56]: This sounds like what's causing the issue I mentioned earlier, and would explain why it sometimes works. Thanks for catching that!

File: src/edits.ts (hunk 2 above)
Comment (boswelja) [reply src/edits.ts:32]: My bad, didn't mean to change it. Will revert asap.

File: README.md (hunk 3 above)
Comment (boswelja) [reply README.md:21]: Have I not updated it correctly? I've renamed releaseFile to releaseFiles and updated the description in the action config already.

File: src/main.ts [hunk 4]
```diff
@@ -51,11 +51,13 @@ async function run() {
             userFractionFloat = undefined
         }
 
-        // Check release file
-        if (!fs.existsSync(releaseFile)) {
-            core.setFailed(`Unable to find release file @ ${releaseFile}`);
-            return
-        }
+        // Check release files
+        releaseFiles.forEach(releaseFile => {
```
Comment (r0adkll) [inline src/main.ts:55]: Please convert this to a `for (in)` loop since the keyword `return` doesn't break a `forEach()` loop

**Review by r0adkll — CHANGES_REQUESTED** [review]
After this it all LGTM

**Comment by boswelja** [comment]
Have you tested this on your end? I seem to be getting random `Error: 0 is invalid`. Wondering if there's an issue with version codes but not sure if it's just me doing something wrong.

**Comment by r0adkll** [comment]
I have not tested it yet, but what output do you get for this line:

```typescript
core.debug(`Creating Track Release for Edit(${appEdit.id}) for Track(${options.track}) with a UserFraction(${options.userFraction}) and VersionCodes(${versionCodes})`);
```

I'm wondering if one of the release uploads is failing and producing a 0 for the version code. I don't actively have any multi-apk projects to test with this so let me know what your output is.

**Comment by boswelja** [comment]
Setting ACTIONS_STEP_DEBUG to true in the repository I'm running it in should show debug output right? I can't get it to print that line :/
The error has however changed to `Unable to find release file @ 0`, so it seems likely an upload is failing. I guess .catch() acts like .forEach() in that it doesn't return where I'm expecting?

**Comment by r0adkll** [comment]
Yes, according to the documentation (https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/enabling-debug-logging#enabling-step-debug-logging) that is how you enable the core.debug() calls to show in the logs. If it's not logging it then my suspicions are that it's not executing.

What does your workflow yaml file look like (more specifically this action config)?

**Comment by boswelja** [comment]
Here's a link to the whole file: https://www.github.com/boswelja/WearableExtensions/tree/main/.github%2Fworkflows%2Fandroid-prod-publish.yml

**Comment by boswelja** [comment]
Solved, and seems to be back to successful runs. Thanks for your support through this, I rarely use Javascript and every time I do I encounter some strange behaviors. Apparently for...in loops iterate over array indices instead of objects, so I've replaced for...in with for...of.

**Review by r0adkll — APPROVED** [review]
:shipit:

**Pull request merged**